tests/types/types.go

```diff
@@ -0,0 +1,205 @@
+// Copyright 2017 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+import (
+	"fmt"
+)
+
+const (
+	RootDeviceImageFile = "blackbox_ignition_test.img"
+)
+
+type File struct {
+	Node
+	Contents string
+	Mode     string
+}
+
+type Directory struct {
+	Node
+	Mode string
+}
+
+type Link struct {
+	Node
+	Target string
+	Hard   bool
+}
+
+type Node struct {
+	Name  string
+	Path  string
```

Comment:
Consider combining `Name` and `Path` for ergonomics, and possibly adding `Name` and `Directory` methods for convenience.  At the very least, `Path` should be renamed to `Dir` or `Directory`, since a file path is always inclusive of its name.